src/jit/lsra.cpp

```diff
@@ -10049,10 +10049,28 @@ regNumber LinearScan::getTempRegForResolution(BasicBlock* fromBlock, BasicBlock*
         if (fromReg != REG_STK)
         {
             freeRegs &= ~genRegMask(fromReg);
+#ifdef _TARGET_ARM_
+            if (type == TYP_DOUBLE && genIsValidFloatReg(fromReg))
+            {
+                regNumber anotherHalfRegNum;
+                anotherHalfRegNum = genIsValidDoubleReg(fromReg) ? REG_NEXT(fromReg) : REG_PREV(fromReg);
+
+                freeRegs &= ~genRegMask(anotherHalfRegNum);
+            }
+#endif
```

Comment:
I suspect that there are other places in resolution that require us to track both halves of the double register. For now, I would suggest adding a wrapper for `genRegMask` that takes, only on ARM, a type, e.g.:
```
getRegisterMask(regNumber reg ARM_ARG( RegisterType type));
```
This would return both bits for a double.

Reply:
> I suspect that there are other places in resolution that require us to track both halves of the double register. 

Yes, as you suspected, I almost found the cause of another remaining bug in resolution conflicts. I will make another PR for that.

> For now, I would suggest adding a wrapper for genRegMask that takes, only on ARM, a type,  on ARM, a type, e.g.:

Thanks for the suggestion. I will apply this.